portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -63,16 +72,24 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       // Get the input from the form.
       String comment = request.getParameter("textInput");
 
+      UserService userService = UserServiceFactory.getUserService();      
```

Comment:
What happens if the user is not logged in here? How can we ensure things fail gracefully (e.g. result in a "you don't have access" error message, rather than a messy stack trace)?